app/src/providerrails/java/com/layer/messenger/flavor/util/CustomEndpoint.java

```
@@ -112,43 +112,48 @@ private static Endpoint getEndpoint() {
         sEndpoints = new HashMap<String, Endpoint>();
 
         // Check for endpoints in resources
-        Context context = App.getInstance();
-        int resId = context.getResources().getIdentifier("layer_endpoints", "raw", context.getPackageName());
-        if (resId == 0) return null;
-
-        // Read endpoints from resources
-        Writer writer = new StringWriter();
-        char[] buffer = new char[1024];
-        InputStream is = context.getResources().openRawResource(resId);
+        Context context = App.getInstance().getApplicationContext();
+//        int resId = context.getResources().getIdentifier("layer_config", "raw", context.getPackageName());
+//        if (resId == 0) return null;
+
         try {
-            Reader reader = new BufferedReader(new InputStreamReader(is, "UTF-8"));
-            int n;
-            while ((n = reader.read(buffer)) != -1) {
-                writer.write(buffer, 0, n);
+            // Read endpoints from resources
+            Writer writer = new StringWriter();
+            char[] buffer = new char[1024];
+            InputStream is = context.getAssets().open("LayerConfiguration.json");
+            try {
+                Reader reader = new BufferedReader(new InputStreamReader(is, "UTF-8"));
+                int n;
+                while ((n = reader.read(buffer)) != -1) {
+                    writer.write(buffer, 0, n);
+                }
+            } catch (Exception e) {
+                if (Log.isLoggable(Log.ERROR)) Log.e(e.getMessage(), e);
```

Crash! Don't ever swallow exceptions :)